File: frontend/src/types/prediction.ts
export interface CategoryPrediction {
  category_id: string
  category_name: string
  predicted_amount: string
}

export interface SpendingPrediction {
  status: 'ok' | 'insufficient_data'
  next_month: string
  predicted_income: string | null
  predicted_expenses: string | null
  by_category: CategoryPrediction[]
}

File: backend/app/schemas/prediction.py
import uuid
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel


class CategoryPrediction(BaseModel):
    category_id: uuid.UUID
    category_name: str
    predicted_amount: Decimal


class SpendingPredictionResponse(BaseModel):
    status: Literal["ok", "insufficient_data"]
    next_month: str
    predicted_income: Decimal | None
    predicted_expenses: Decimal | None
    by_category: list[CategoryPrediction]
